nixnet/system/databases.py

```diff
@@ -0,0 +1,147 @@
+﻿from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import itertools
+import six
+import typing  # NOQA: F401
+
+from nixnet import _funcs
+
+
+class Databases(collections.Mapping):
+    """Database aliases."""
+
+    def __init__(self, handle):
+        self._handle = handle
+
+    def __repr__(self):
+        return 'System.Databases(handle={0})'.format(self._handle)
+
+    def __len__(self):
+        return len(_funcs.nxdb_get_database_list(''))
+
+    def __iter__(self):
+        for alias, filepath in _funcs.nxdb_get_database_list(''):
+            yield alias
+
+    def __getitem__(self, index):
+        # type: (str) -> Database
+        """Return the Database object associated with the specified alias.
+
+            Args:
+                index(str): The value of the index (alias name).
+        """
+        if isinstance(index, six.string_types):
+            for alias, filepath in _funcs.nxdb_get_database_list(''):
+                if alias == index:
+                    return self._create_item(self._handle, index, alias, filepath)
+            else:
+                raise ValueError('Database alias %s not found in the system' % index)
+        else:
+            raise KeyError(index)
```

Comment:
I might not have been clear before.

We should throw `KeyError` when the key isn't found.

As for what to throw on unsupported types, it looks like we've set the precedence of `TypeError`
https://github.com/ni/nixnet-python/blob/master/nixnet/_session/collection.py#L44